asv/config.py

```diff
@@ -2,13 +2,44 @@
 
 import os
 import sys
+from pathlib import Path
 
 from .console import log
 from . import util
 
 # TODO: Some verification of the config values
 
 
+def _get_config_path():
+    """
+    Check the default config file path for all valid extensions.
+    Raise if no file is found, or if multiple are found for the valid extensions.
+    Return the path of the config file if exactly one is found.
+    """
+    extensions = ['.json', '.jsonc']
+    path = Path('asv.conf')
+
+    num_matches = 0
+    for e in extensions:
+        p = path.with_suffix(e)
+        if p.exists:
+            num_matches += 1
+            path = p
+
+    if num_matches == 0:
+        raise util.UserError(
+            f"No `asv.conf` file found for valid extensions: {extensions}."
+        )
+    elif num_matches > 1:
+        raise util.UserError(
+            f"Multiple `asv.conf` files found for valid extensions: {extensions}. "
+            f"Please specifiy which file to use with `--config=FILEPATH`."
+        )
+    else:
+        # exactly one valid config file found
+        return path
```

Comment:
Might need `str(path)`

Reply:
Yup I think `asv` expects path, but otherwise this looks great

Reply:
```suggestion
        return str(path)
```